Sample admin (#17)
inbo/n2khab-sample-admin: src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd

> # Generate mhq sample administration table

Reviewer: General remark about the subparagraphs in this part (e.g. below 'Samples assigned to INBO'): while the textual explanations are reasonable, it is still difficult to follow what is actually done and what the outcome is, if the resulting (intermediate) objects are not somehow visualised (e.g. with `glimpse()`, first rows of a table etc.). While such visualisation is done to enlist anomalies, it isn't for the main data workflow. This makes it uneasy to understand what is meant with titles like '_Combine sample table with admin data_', '_Combine sample admin data and assessments data_' etc. without running code in an interactive session and inspecting results. In such cases the accompanying text focuses on specific aspects, not on explanation of the actual step (it's my impression, maybe partly wrong).

#### Samples for which the subtype of the observed type is unknown

```{r}

check_subtype <- sample_assessments_anb %>%
    filter(!is.na(type_observed)) %>%
    filter(type_observed %in% types$main_type) %>%
    filter(type_observed %in% subtypes$main_type)
```

#### Observed type was not assigned to ANB or no sample set was drawn for observed type

```{r}

check_anb_type <- sample_assessments_anb %>%
    filter(assessment_status == "in situ assessment") %>%
    filter(measured) %>%
    filter(! str_sub(type_observed, 1 , 4) %in% main_type_anb) %>%
    select(sample_id, type_target, type_observed)

write.table(check_anb_type, "../../output/check_anb_type.txt")

check_anb_type %>% 
    kable() %>%
    kable_styling()

```

## Combine ANB and INBO admin table

```{r}
sample_assessments <- bind_rows(
    sample_assessments_anb,
    sample_assessments_inbo
) %>%
    mutate(type_target = factor(type_target, levels= levels(types$type)),
           assessment_status = factor(assessment_status, levels = c("in situ assessment", "orthophoto assessment", "temporary inaccessable", "permanently inaccesable", "no assessment")),
           sample_source = "habitatmap")
```

> ## Create new samples when is_type_target = FALSE and allow_other_type = TRUE

Reviewer: ```suggestion
## Create new sampling units when is_type_target = FALSE and allow_other_type = TRUE
```

We still have to decide what to do with samples that were measured for another type than the target type and for which the location was changed.

Do we have to document that these are new samples compared to the original set of samples? How?

```{r}
new_samples <- sample_assessments %>%
    filter(assessment_status == "in situ assessment" & !is.na(assessment_status)) %>%
    filter((!is_type_target) & allow_other_type) %>%
    mutate(sample_id = str_c(location_id, "_", type_observed),
           type_target = factor(type_observed, levels = levels(types$type)),
           is_type_target = TRUE,
           allow_other_type = NA,
           sample_source = "mhq assessment") %>%
    select(-year_planned) %>%
    unique()

```

> ## Add new samples to sample admin table

Reviewer: ```suggestion
## Add new sampling units to sample admin table
```

```{r}
sample_assessments_update <- bind_rows(
    sample_assessments,
    new_samples
) %>% 
    arrange(type_target, sac, grts_ranking)

```

To do: check grts-ranking of dune samples.

## Handle location changes

How do we handle location changes? Do we create a new sample_id and do we set in the orignal sample_id is_type_target = FALSE? 

## Export

### Overview of all assessed samples

```{r}
mhq_assessments_summary <- sample_assessments_update %>%
    filter(!is.na(assessment_status)) %>%
    filter(assessment_status != "no assessment") %>%
    filter(sample_source != "mhq_assessment") %>%
    filter(!is.na(is_type_target)) %>% # sommige inbo measured plots zijn nog niet ingevoerd in inboveg
    select(sample_id,  grts_ranking, assessment_status, type_target, is_type_target, type_observed, location_change, distance_from_sample, x_orig, y_orig, x_measured, y_measured)
   
check <- n_distinct(mhq_assessments_summary$sample_id) == nrow(mhq_assessments_summary)
check_n <- mhq_assessments_summary %>%
    group_by(sample_id) %>%
    mutate(n = n()) %>%
    ungroup()

> write_vc(mhq_assessments_summary, root = "../../n2khab_mhq_data" , file = "mhq_assessments", sorting = c("type_target", "grts_ranking"))

Reviewer: Added selected preview as a reference for discussion:

```r
> mhq_assessments %>% slice(seq(1,1401,100)) %>% knitr::kable()
```

|sample_id         | grts_ranking|assessment_status  |type_target |is_type_target |type_observed |location_change | distance_from_sample|   x_orig|   y_orig| x_measured| y_measured|
|:-----------------|------------:|:------------------|:-----------|:--------------|:-------------|:---------------|--------------------:|--------:|--------:|----------:|----------:|
|1338805_1330_da   |      1338805|in situ assessment |1330_da     |TRUE           |1330_da       |FALSE           |                  4.9|  61661.6| 221902.1|    61666.0|   221899.9|
|1781173_1330_hpr  |      1781173|in situ assessment |1330_hpr    |FALSE          |NA            |NA              |                   NA|  62301.6| 222030.1|         NA|         NA|
|1036081_2120      |      1036081|in situ assessment |2120        |FALSE          |NA            |NA              |                   NA|  34173.6| 204462.1|         NA|         NA|
|16772913_2190_mp  |     16772913|in situ assessment |2190_mp     |TRUE           |2190_mp       |TRUE            |                  0.0|  32733.6| 203598.1|    32733.6|   203598.1|
|359574_2330_bu    |       359574|in situ assessment |2330_bu     |FALSE          |2310          |FALSE           |                  0.1| 168765.6| 205998.1|   168765.6|   205998.2|
|1762774_2330_bu   |      1762774|in situ assessment |2330_bu     |TRUE           |2330_bu       |FALSE           |                  0.1| 185661.6| 211054.1|   185661.6|   211054.1|
|120530_4030       |       120530|in situ assessment |4030        |TRUE           |4030          |FALSE           |                  0.1| 224765.6| 186926.1|   224765.6|   186926.2|
|2181769_6230      |      2181769|in situ assessment |6230        |FALSE          |NA            |NA              |                   NA| 112029.6| 193550.1|         NA|         NA|
|753630_6230_hmo   |       753630|in situ assessment |6230_hmo    |TRUE           |6230_hmo      |TRUE            |                 31.3| 160157.6| 225582.1|   160126.4|   225579.4|
|912594_6230_hn    |       912594|in situ assessment |6230_hn     |FALSE          |NA            |NA              |                   NA| 224445.6| 194670.1|         NA|         NA|
|7072966_6410_mo   |      7072966|in situ assessment |6410_mo     |TRUE           |6410_mo       |FALSE           |                  0.3| 162429.6| 179534.1|   162429.7|   179534.4|
|4599942_6410_ve   |      4599942|in situ assessment |6410_ve     |TRUE           |6410_ve       |TRUE            |                 50.3| 179069.6| 184846.1|   179068.8|   184795.8|
|1052582_6510_hu   |      1052582|in situ assessment |6510_hu     |TRUE           |6510_hu       |FALSE           |                  0.1| 185885.6| 181038.1|   185885.6|   181038.1|
|10622226_6510_hus |     10622226|in situ assessment |6510_hus    |TRUE           |6510_hus      |FALSE           |                  0.0| 219133.6| 217454.1|   219133.6|   217454.1|
|4322230_7140_meso |      4322230|in situ assessment |7140_meso   |FALSE          |NA            |NA              |                   NA| 218269.6| 198830.1|         NA|         NA|

# Generate mhq sample administration table

In the mhq sample adminstration table we want to document assessments and measurements of the mhq samples. 
An assessment of a mhq sample is seen as evaluation whether or not a sample is valid. 

> A mhq sample is valid if the target type (= the type for which the sample was selected) corresponds with the observed type in the center of the grid cell. 

Author: see response to general remarks

### Determine type in plot center and check if the observed type corresponds with the target type

Vegetation type cover is measured both in the square plot and the circle plot. 
The field protocol prescribes that a plot is only recorded when the plot center is located within (one of) the target habitat type(s). 
When the observed type is different from (one of) the target type(s), the plot can be recorded when the GRTS ranking of the sample location is smaller than the largest GRTS ranking in the set of samples for the observed type. 

In some cases the subtype is not specified. When this is the case, we assume that the subtype of the observed type corresponds to the target type.

> Some recorded plots contain different habitat types or other cover types. 

Author: land cover/ land use types (for example 'akker')

> Some recorded plots contain different habitat types or other cover types. 

Reviewer: OK, best clarify this in the text

However it is not explicitely recorded which of the cover types is located in the center of the plot. 
Yet this is essential information to determine if a sample is valid. 
Therefore, when a plot contains different cover types we assume that the plot center is located in the type recorded in the square plot that corresponds to (one of) the target type(s). 

> When none of the target habitat types are observed we assume that the plot center is located in the type that belongs to the same main type as (one of) the target habitat types. We apply these rules and check of we select one type for each plot this way.

Author: For example: the target types in the reference point are 4010 and 6510_hu. the observed types in the plot are 6510_hus and 6210. then we assume that 6510_hus is the observed type in the reference point as it belongs to the main type 6510, which is the main type of one of the target types.

> When none of the target habitat types are observed we assume that the plot center is located in the type that belongs to the same main type as (one of) the target habitat types. We apply these rules and check of we select one type for each plot this way.

Reviewer: OK, I see what you mean now. The confusion came from "... the type that belongs to the same main type as ...". From "When none of the target habitat types are observed" it is not evident that _another subtype from a same_ (targetted) main type _is_ observed, which is the case you handle. Best describe the handled circumstance more clearly.

Do you describe what happens in the following cases then (perhaps they don't occur?), where 4010 and 6510_hu are target types?

1. 6210 and 6410_mo were observed 
2. 4010 and 6510_hu (the target types) were observed

> When none of the target habitat types are observed we assume that the plot center is located in the type that belongs to the same main type as (one of) the target habitat types. We apply these rules and check of we select one type for each plot this way.

Author: OK. I clarified it by using an example. 
Unfortunately, the two other cases you describe do occur (only a couple of cases). In case of 1. we select the type with the highest cover. In case of 2. we report it to the fieldwork coordinator to clarify.

> When none of the target habitat types are observed we assume that the plot center is located in the type that belongs to the same main type as (one of) the target habitat types. We apply these rules and check of we select one type for each plot this way.

Reviewer: OK, best distinguish between those cases in the text

> When none of the target habitat types are observed we assume that the plot center is located in the type that belongs to the same main type as (one of) the target habitat types. We apply these rules and check of we select one type for each plot this way.

Author: OK I did

Vegetation type cover is measured both in the square plot and the circle plot. 
The field protocol prescribes that a plot is only recorded when the plot center is located within (one of) the target habitat type(s). 

> When the observed type is different from (one of) the target type(s), the sampling unit can be measured when grts_ranking_draw of the reference point is smaller than the largest rts_ranking_draw in the sample of the observed type. 

Reviewer: ```suggestion
When the observed type is different from (one of) the target type(s), the sampling unit can be measured when grts_ranking_draw of the reference point is smaller than the largest grts_ranking_draw in the sample of the observed type. 
```

Author: OK

    st_drop_geometry() %>%
    mutate(x = x_assessment,
           y = y_assessment)

#check: no lsvi measurement AND location change

check <- sample_assessments_update %>%
    filter(distance_from_refpoint > 1) %>%
    filter(!lsvi_measurement)

sample_units_update_orig_location <- sample_assessments_update %>%
    filter(distance_from_refpoint > 1) %>%
    mutate(is_type_target = FALSE,
           location_change = NA,
           lsvi_measurement = FALSE,
           completed = NA,
           x = x_orig,
           y = y_orig,
           is_valid = FALSE,
           allow_other_type = NA,
           type_observed = NA,
           type_cover_square = NA,
           type_cover_circle = NA,
           distance_from_refpoint = 0)
    
sample_assessments_update2 <- sample_assessments_update %>%
    filter(distance_from_refpoint < 1 | is.na(distance_from_refpoint)) %>%
    mutate(x = ifelse(is.na(x_assessment), x_orig, x_assessment),
           y = ifelse(is.na(y_assessment), y_orig, y_assessment)) %>%
    bind_rows(sample_units_change_location) %>%
    bind_rows(sample_units_update_orig_location) %>%
    arrange(point_code)
```

 

## Export

### Update mhq_terr_refpoints

> The new sampling units/reference points are added to the file 'mhq_terr_refpoints'. For now, we give the updated file a new name: 'mhq_terr_refpoints_check_update'.   

Author: see response to general remarks

    arrange(type, sampling_unit_code)

write_vc(mhq_terr_refpoint_update, file = "mhq_terr_refpoints_details_check_update", root = "../../data", sorting = c("type", "sampling_unit_code", "point_code"), strict = FALSE)
```

### Create mhq_terr_refpoints_validity

The table mhq_terr_refpoints_validity contains following variables:

- sampling_unit_code
- point_code
- date
- is_valid

The variabe is_valid determines if a sampling unit is valid for mhq at the location corresponding to point_code.

The table below shows some records as an example.

```{r}
mhq_refpoints_validity <- sample_assessments_update2 %>%
    filter(!is.na(is_valid)) %>%
    filter(!is.na(scheme)) %>%
    select(scheme, sampling_unit_code, point_code, date = assessment_year, is_valid)

write_vc(mhq_refpoints_validity, file = "mhq_terr_refpoints_validity", root = "../../data", sorting = c("scheme", "sampling_unit_code", "point_code"))
```

```{r}
mhq_refpoints_validity %>%
    slice(1000:1009)
```

### Create mhq_terr_assessments

The table mhq_terr_assessments gives an overview of all assessments (see below for example records). 

```{r}
mhq_terr_assessments <- sample_assessments_update2 %>%

>     select(point_code, type_target, is_type_target, type_observed, inaccessable, assessment_date = assessment_year, assessment_source, x, y)

Author: OK I changed this. Now assessment_date  is a date variable. When only the assessment year is known I create a dummy date as you suggest.
